File: test-suite/src/test/java/io/micronaut/picocli/docs/cli/app/MyCliAppCommand.java
package io.micronaut.picocli.docs.cli.app;

// tag::imports[]
import io.micronaut.configuration.picocli.PicocliRunner;
import org.slf4j.Logger;
import picocli.CommandLine.Command;
import picocli.CommandLine.Option;

import static org.slf4j.LoggerFactory.getLogger;
// end::imports[]

// tag::class[]
@Command(name = "my-cli-app", description = "...", mixinStandardHelpOptions = true) // <1>
public class MyCliAppCommand implements Runnable { // <2>
    private static final Logger LOG = getLogger(MyCliAppCommand.class);

    @Option(names = {"-v", "--verbose"}, description = "...") // <3>
    boolean verbose;

    public static void main(String[] args) throws Exception {
        PicocliRunner.run(MyCliAppCommand.class, args); // <4>
    }

    public void run() { // <5>
        // business logic here
        if (verbose) {
            LOG.info("Hi!");
        }
    }
}
// end::class[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/picocli/docs/cli/app/MyCliAppCommand.kt
package io.micronaut.picocli.docs.cli.app

// tag::imports[]
import io.micronaut.configuration.picocli.PicocliRunner
import org.slf4j.LoggerFactory
import picocli.CommandLine.Command
import picocli.CommandLine.Option
// end::imports[]

// tag::class[]
@Command(name = "my-cli-app", description = ["..."], mixinStandardHelpOptions = true) // <1>
class MyCliAppCommand : Runnable { // <2>

    @Option(names = ["-v", "--verbose"], description = ["..."]) // <3>
    var verbose = false

    companion object {
        private val LOG = LoggerFactory.getLogger(MyCliAppCommand::class.java)

        @Throws(Exception::class)
        @JvmStatic
        fun main(args: Array<String>) {
            PicocliRunner.run(MyCliAppCommand::class.java, *args) // <4>
        }
    }

    override fun run() { // <5>
        // business logic here
        if (verbose) {
            LOG.info("Hi!")
        }
    }
}
// end::class[]
